docs/user/tutorials/devcontainer.rst

```diff
@@ -0,0 +1,73 @@
+
+Setup the Devcontainer
+======================
+
+Introduction
+------------
+
+The devcontainer provides the environment in which you will do all your development.
+
+
+Configure Visual Studio Code
+----------------------------
+
+You must ensure you have the devcontainer plugin:
+
+
+.. figure:: ../images/dev-container-plugin.png
+    :width: 600px
+    :align: center
+
+    devcontainer plugin
+
+
+For podman users, you must first tell VSCode to use podman instead of docker.
+Open a VSCode window and hit "ctrl ," (control-comma) to open the user
+settings editor and search for
+"dev.containers.dockerPath", change its value from "docker" to "podman".
+
+
+.. figure:: ../images/dev-container-settings.png
+    :width: 600px
+    :align: center
+
+    devcontainer podman setting
+
+
+Launching the Devcontainer
+--------------------------
+
+To setup your devcontainer, perform the following steps:
+
+-  create a workspace folder
+-  clone the .devcontainer repository into the workspace folder
+-  open the workspace folder with Visual Studio Code.
+
+for example:
+
+.. code-block:: bash
+
+    mkdir work-ec
+    cd work-ec
+    git clone git@github.com:epics-containers/.devcontainer.git
```

Comment:
This is fine for epics-containers but if you want this to be a generic document for users of python-pip-skeleton then it is not clear what to do for those users.

We have two slightly different setups

- epics-containers has a devcontainer called dev-e7 but we clone a .devcontainer project that referes to that in its Dockerfile and sits beside the other projects that will make up your workspace
- python3-pip-skeleton projects have their .devcontainer folder embedded so you only clone the parent project and vscode picks up the subfolder.

This is not rocket science but does sound like a lot to explain for a new starter looking to this as entry level docs.

Lets discuss in person.